src/main/java/com/zufar/onlinestore/payment/endpoint/PaymentEndpoint.java

```diff
@@ -38,36 +36,63 @@ public class PaymentEndpoint {
     private final PaymentApi paymentApi;
 
     @PostMapping
-    public ResponseEntity<PaymentDetailsWithTokenDto> createPayment(@RequestBody @Valid final CreatePaymentDto createPaymentDto) throws PaymentIntentProcessingException {
-        PaymentDetailsWithTokenDto createdPayment = paymentApi.createPayment(createPaymentDto);
+    public ResponseEntity<ApiResponse<ProcessedPaymentWithClientSecretDto>> processPayment(@RequestParam @NotEmpty final String cardDetailsTokenId) {
+        ProcessedPaymentWithClientSecretDto processedPayment = paymentApi.processPayment(cardDetailsTokenId);
+
+        ApiResponse<ProcessedPaymentWithClientSecretDto> apiResponse = ApiResponse.<ProcessedPaymentWithClientSecretDto>builder()
+                .data(processedPayment)
+                .messages(List.of("Payment successfully processed"))
+                .timestamp(LocalDateTime.now())
+                .httpStatusCode(HttpStatus.CREATED.value())
+                .build();
+
         return ResponseEntity.status(HttpStatus.CREATED)
-                .body(createdPayment);
+                .body(apiResponse);
     }
 
     @GetMapping("/{paymentId}")
-    public ResponseEntity<PaymentDetailsDto> getPaymentDetails(@PathVariable @NotNull final Long paymentId) throws PaymentNotFoundException {
-        PaymentDetailsDto retrievedPayment = paymentApi.getPaymentDetails(paymentId);
-        log.info("Get payment details: payment details: {} successfully retrieved.", retrievedPayment);
-        return ResponseEntity.ok()
-                .body(retrievedPayment);
-    }
+    public ResponseEntity<ApiResponse<ProcessedPaymentDetailsDto>> getPaymentDetails(@PathVariable @NotNull final Long paymentId) {
+        ProcessedPaymentDetailsDto retrievedPayment = paymentApi.getPaymentDetails(paymentId);
 
-    /**
-     * This endpoint is used only until we have an implementation of this logic on the frontend side.
-     * It will come in handy for testing the API.
-     */
-    @PostMapping("/method")
-    public ResponseEntity<String> getPaymentMethod(@RequestBody @Valid final CreatePaymentMethodDto createPaymentMethodDto) throws PaymentMethodProcessingException {
-        String paymentMethodId = paymentApi.createPaymentMethod(createPaymentMethodDto);
-        return ResponseEntity.ok()
-                .body(paymentMethodId);
+        ApiResponse<ProcessedPaymentDetailsDto> apiResponse = ApiResponse.<ProcessedPaymentDetailsDto>builder()
+                .data(retrievedPayment)
+                .messages(List.of("Payment successfully retrieved"))
```

Comment:
I think we should use messages only for errors. Just leave messages empty as you have data field filled by `retrievedPayment`

Reply:
Done